short-read-mngs/host_filter.wdl

```diff
@@ -1,674 +1,619 @@
 version 1.0
 
-task RunValidateInput {
+# CZ ID short-read-mngs pipeline stage 1 (2022 version):
+# - input validation & QC
+# - host & human filtering
+# - deduplication
+# - subsampling
+workflow czid_host_filter {
   input {
+    File fastqs_0
+    File? fastqs_1
+    String nucleotide_type = "DNA"
+
+    File adapter_fasta
+
+    String host_genome
+    File bowtie2_index_tar
+    File hisat2_index_tar
+    File kallisto_idx
+
+    File human_bowtie2_index_tar
+    File human_hisat2_index_tar
+
+    Int max_input_fragments
+    Int max_subsample_fragments
+
+    Int cpu = 16
     String docker_image_id
+
+    # legacy idseq-dag inputs:
+    String file_ext = "fastq"
     String s3_wd_uri
-    Array[File] fastqs
-    Int max_input_fragments
-    String file_ext
-  }
-  command<<<
-  set -euxo pipefail
-  idseq-dag-run-step --workflow-name host_filter \
-    --step-module idseq_dag.steps.run_validate_input \
-    --step-class PipelineStepRunValidateInput \
-    --step-name validate_input_out \
-    --input-files '[["~{sep='","' fastqs}"]]' \
-    --output-files '["validate_input_summary.json", ~{if length(fastqs) == 2 then '"valid_input1.fastq", "valid_input2.fastq"' else '"valid_input1.fastq"'}]' \
-    --output-dir-s3 '~{s3_wd_uri}' \
-    --additional-files '{}' \
-    --additional-attributes '{"truncate_fragments_to": ~{max_input_fragments}, "file_ext": "~{file_ext}"}'
-  >>>
-  output {
-    String step_description_md = read_string("validate_input_out.description.md")
-    File validate_input_summary_json = "validate_input_summary.json"
-    File valid_input1_fastq = "valid_input1.fastq"
-    File? valid_input2_fastq = "valid_input2.fastq"
-    File? output_read_count = "validate_input_out.count"
-    File? input_read_count = "fastqs.count"
   }
-  runtime {
-    docker: docker_image_id
+
+  # Validate input reads (and truncate if very large)
+  call RunValidateInput {
+    input:
+    reads1_fastq = fastqs_0,
+    reads2_fastq = fastqs_1,
+    file_ext = file_ext,
+    max_input_fragments = max_input_fragments,
+    docker_image_id = docker_image_id,
+    s3_wd_uri = s3_wd_uri
   }
-}
 
-task RunStar {
-  input {
-    String docker_image_id
-    String s3_wd_uri
-    File validate_input_summary_json
-    Array[File] valid_input_fastq
-    File star_genome
-    String nucleotide_type
-    String host_genome
-    String genome_dir = "STAR_genome/part-0/"
+  # Adapter trimming and QC filtering
+  call fastp_qc {
```

Comment:
nit: can we rename the steps slightly? I believe our step names are in camelCase and start with a verb

Reply:
e.g. `fastp_qc` => `RunFastpQc`